src/Raven.Server/Documents/Indexes/MapReduce/ReduceKeyProcessor.cs

```diff
@@ -217,15 +199,37 @@ public void Process(object value)
                 return;
             }
 
-            var dynamicArray = value as DynamicArray;
+            var dynamicJson = value as DynamicBlittableJson;
 
-            if (dynamicArray != null)
+            if (dynamicJson != null)
             {
+                var obj = dynamicJson.BlittableJson;
+
+                _mode = Mode.MultipleValues;
+
                 if (_buffer == null)
                     _buffer = _buffersPool.Allocate(16);
+                
+                for (int i = 0; i < obj.Count; i++)
+                {
+                    // this call ensures properties to be returned in the same order, regardless their storing order
```

Comment:
This will actually iterate over them in the store index order. Which is good, that is what we want.
Different order is actually different object as far as we are concerned (remember that they are being generated by the same function, guaranteed to be the same).

But what made you think that this will be in some sort of order other than the store order?